### DISCUSSION TITLE
Using FloorPlanner instead of Sweet Home 3D - how do i make the lighting show?

### DISCUSSION THREAD

**Discussion by magicmega** [ExperienceLovelace/ha-floorplan#346] [Q&A]
A long time ago in a different house, I created a full floorplan for HomeAssistant with SweetHome 3D.

This time I saw the docs said to use Floorplanner so I did that. I overdid it a bit, making it hyper realistic.

However, I'm at the point of exporting the image and starting the SVG work in Inkscape and I'm stuck.
- In SweetHome 3D, the lights are dynamic - you can turn them all on and take a render with that view
- In Floorplanner, the lights seem to act just like any other piece of furniture and don't seem to have the ability to actually show their lighting.

How are people creating their floorplan images when using Floorplanner? Is it just a simple remove ceilings and blast with daylight and take that one image? I'd much rather be able to show curved edges for the lighting reach on individual lights like you could with SweetHome 3D. Am I missing something?

**Comment by exetico** [comment]
I'm about to give you a bit of a sad information:
Most people only use it for a single static background, like me. With that said, I've personally though about rendering lights individuelly, but you'er right... Its not possible to have it auto-rendered by Floorplanner.

Hopefully other users has better experince with this. 
Personally I just add the dimmed layer on top, if the light is turned on, but that doesnt give any light bleeds, for sure.

**Reply by Alfiegerner** [reply]
I went through a lot of effort to take images with or without lights, did the images in GIMP and then apply the light 'diff'. Gets hard with more than one light in a room, you have to take multiple 'on'images then use some custom JavaScript to select which version of combined lights is on, and therefore which 'on' overlay of lights you need . Custom JavaScript can also pull off RGB overlay too

Very well might be an easier way, I think using picture elements you can use CSS effects for lighting more easily but can't remember why I came to that conclusion.

https://www.reddit.com/r/homeassistant/comments/geawqn/floorplan_using_lovelacefloorplan/?utm_source=share&utm_medium=android_app&utm_name=androidcss&utm_term=1&utm_content=2

**Reply by OzGav** [reply]
Yes I didn’t bother with trying to do multiple images of all the light combinations. I have the opacity of the top room light layer at around 65% and I use additional icons to control the lamps in the room as well as show their state more clearly.
![IMG_0830](https://github.com/ExperienceLovelace/ha-floorplan/assets/19848947/3a1b713b-b2a8-48d0-ad1c-39a6c4e67ba2)
